### PR TITLE
JBERET-529 Make it possible to limit number of execution records retr…

### PR DESCRIPTION
**Pull request by TomasHofman**
…ieved by a job repository

https://issues.redhat.com/browse/JBERET-529
https://issues.redhat.com/browse/WFLY-15525

### CODE REVIEW

**Comment by TomasHofman** [comment]
Added test cases.

File: jberet-core/src/main/java/org/jberet/repository/JdbcRepository.java [hunk 1]
```diff
@@ -1061,11 +1067,20 @@ public void executeStatements(final String statements, final String statementsRe
...
             while (rs.next()) {
                 final long i = rs.getLong(1);
                 result.add(i);
+                count++;
+                if (maxRecords != null && count >= maxRecords) {
+                    BatchLogger.LOGGER.jobExecutionRecordsLimited(maxRecords);
+                    break;
+                }
             }
+            BatchLogger.LOGGER.debugf("Number of job execution records read: %d", result.size());
```
Comment (chengfang) [inline jberet-core/src/main/java/org/jberet/repository/JdbcRepository.java:1083]: can we add this into a separate loop for a non-null `maxRecords`?  Most users probably won't bother to configure this property. Something like:

```java
if(null) {
  //old while loop
  while (...) {...}
} else {
 //new loop
 count = 0;
 while(rs.next() && count < maxRecords) {
    ...
 }
 log(...)
}

File: jberet-core/src/main/java/org/jberet/_private/BatchLogger.java [hunk 2]
```diff
@@ -158,4 +158,8 @@
     @Message(id = 33, value = "Stopping job execution: %s")
     @LogMessage(level = Logger.Level.INFO)
     void stoppingJobExecution(long jobExecutionId);
+
+    @Message(id = 34, value = "The number of job executions read from the job repository is limited to %d.")
+    @LogMessage(level = Logger.Level.DEBUG)
+    void jobExecutionRecordsLimited(int limit);
 }
```
[the remark this answers is not in the document]
Comment (TomasHofman) [reply jberet-core/src/main/java/org/jberet/_private/BatchLogger.java:165]: Done.

File: jberet-core/src/main/java/org/jberet/repository/JdbcRepository.java (hunk 1 above)
Comment (TomasHofman) [reply jberet-core/src/main/java/org/jberet/repository/JdbcRepository.java]: Sure, I updated the PR.

**Review by chengfang — APPROVED** [review]

**Comment by chengfang** [comment]
@TomasHofman can you also apply the changes to 1.3.x and 1.4.x branches?

**Comment by TomasHofman** [comment]
@chengfang will do. Thanks!

**Comment by TomasHofman** [comment]
Downstream PRs are linked.

**Pull request merged**